src/api/api.service.ts

```diff
@@ -32,89 +66,87 @@ export class ApiService {
         );
       }
       return userApis;
-    }
-    catch(error) {
+    } catch (error) {
       throw new BadRequestException(
         ZaLaResponse.BadRequest('Internal Server error', error.message, '500'),
       );
     }
   }
 
-    async createApi(createApiDto: CreateApiDto, profileId: string) {
-        try {
-            const apiExist = await this.apiRepo.findOne({
-                where: { name: createApiDto.name, }
-            });
 
-            if (apiExist) {
-                throw new BadRequestException(
-                    ZaLaResponse.BadRequest(
-                        'Existing values',
-                        'An api with with this name already exist... try another name',
-                    ),
-                );
-            }
-            const uniqueApiSecretKey = uuid();
-            const newApi = this.apiRepo.create({
-                ...createApiDto,
-                profileId,
-                secretKey: uniqueApiSecretKey,
-            });
-            const saveApi = await this.apiRepo.save(newApi);
-            return saveApi;
-        }
-        catch(err) {
-            throw new BadRequestException(
-                ZaLaResponse.BadRequest('Internal Server error', err.message, '500'),
-            );
-        }
+  /**
+   * @param {string} apiId - id of the API
+   * @param {string} profileId -  id of the user who is trying to get the api.
+   * checks if the api exist and the user is can get the api.
+   * @returns The getAnApi method returns a promise of unique api the user(profileId) requested .
+   */
+  
+  async getAnApi(apiId: string, profileId: string) {
+    const api = await this.apiRepo.findOne({ where: { id: apiId } });
+    if (!api) {
+      throw new NotFoundException(
+        ZaLaResponse.NotFoundRequest(
+          'Not found',
+          'The api does not exist',
+          '404',
+        ),
+      );
     }
+    if (api.profileId !== profileId) {
+      throw new NotFoundException(
+        ZaLaResponse.NotFoundRequest(
+          'Not found',
```

Comment:
There is no need for this. You only have to get a single api by id.

Reply:
the reason i added that is to confirm if the profileId has access to the api